packages/eslint-plugin/tests/rules/consistent-type-assertions.test.ts

```diff
@@ -41,8 +41,8 @@ const x = !'string' as A;
 const x = (a as A) + b;
 const x = (a as A) + (b);
 const x = new Generic<string>() as Foo;
-const x = new (Generic<string> as Foo)();
-const x = new (Generic<string> as Foo)('string');
+const x = new ((Generic<string>) as Foo)();
+const x = new ((Generic<string>) as Foo)('string');
```

Comment:
From playing around with the AST viewer, I think this test case was simply incorrect (though tbh this syntax is bonkers and I struggle to follow it so I'm not totally certain.)

For reference, the inputs are 
```typescript
const x = new (<Foo>Generic<string>)();
const x = new (<Foo>Generic<string>)('string');
```
(located above in the ANGLE_BRACKET_TESTS_EXCEPT_CONST_CASE string.)